### PR TITLE
[RF] Avoid compile time dependency of RooFit on RDataFrame

### PR DESCRIPTION
**Pull request by guitargeek**
For better modularity in the future, the major end-user ROOT components
like RDataFrame, RooFit, and TMVA should not depend on each other.

There was a dependency of RooFit on RDataFrame because of the
RooAbsDataHelper, but it is actually easy to avoid this with forward
declarations.

This should have no impact on the user, because if they use RDF, they
probably include it already if they're not using the interpreter.

### CODE REVIEW

**Review by vepadulano — COMMENTED** [review]
Some generic comments before going through the code itself:

* If this class is to be moved to `tree/dataframe`, then I believe we should remove `roofit/RDataFrameHelpers` directory altogether (in the long run, see next point).
* I don't believe we can change the header before a release without a transition path. I think we should start by including a compile-time warning whenever a user `#include`s the `RooAbsDataHelper.h` header, telling them that they should instead include `ROOT/RooAbsDataHelper.hxx`. This can be a simple forwarding header. Then in the next ROOT release (or the one after), we can completely remove the header together with the whole `roofit/RDataFrameHelpers` directory.

**Comment by guitargeek** [comment]
Thanks for the comments! I will change things a little bit to address these concerns.

What made me weary now is that the compiler doesn't suggest any `override` keywords for the RDF-related member functions in the RooAbsDataHelper. So what I think is that it's probably not even necessary to inherit from `RActionImpl`! Maybe that was just done by the original author for convenience. I think this dependency can actually be avoided without changing and RDF code.

**Comment by guitargeek** [comment]
@vepadulano, crazy, it's even easier than I thought! It's working in C++ to forward-declare the bases of template classes :+1:

So I don't need to change any RDF code. Note that the `RooAbsDataFiller` header that I remove now was an implementation detail that was only introduced in this development cycle, so no problem to remove it. The `RooAbsDataHelper.h` is also staying, it just lives in RooFitCore.

File: roofit/roofitcore/inc/RooAbsDataHelper.h [hunk 1]
```diff
@@ -10,15 +10,74 @@
...
+class RooAbsDataFiller {
+public:
+   RooAbsDataFiller();
+
+   /// Move constructor. It transfers ownership of the internal RooAbsData object.
+   RooAbsDataFiller(RooAbsDataFiller &&other) : _events{std::move(other._events)}, _eventSize{other._eventSize} {}
+
+   /// Copy is discouraged.
+   /// Use `rdataframe.Book<...>(std::move(absDataHelper), ...)` instead.
+   RooAbsDataFiller(const RooAbsDataFiller &) = delete;
```
Comment (vepadulano) [inline roofit/roofitcore/inc/RooAbsDataHelper.h:40]: I think we're missing the rule-of-five methods here

**Pull request merged**